iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/VoltageAngleLimitAdderImpl.java

```diff
@@ -71,19 +67,23 @@ public VoltageAngleLimit add() {
         if (id == null) {
             throw new IllegalStateException("Voltage angle limit id is mandatory.");
         }
-        if (subnetworkId != null && checkTerminalsInSubnetwork(subnetworkId)) {
+        if (subnetwork != null && checkTerminalsInSubnetwork(subnetwork)) {
             throw new ValidationException(this, "The involved voltage levels are not in the subnetwork '" +
-                    subnetworkId + "'. Create this VoltageAngleLimit from the parent network '" + networkRef.get().getId() + "'");
+                    subnetwork + "'. Create this VoltageAngleLimit from the parent network '" + network.getId() + "'");
         }
-        if (networkRef.get().getVoltageAngleLimitsIndex().containsKey(id)) {
-            throw new PowsyblException("The network " + networkRef.get().getId()
+        if (network.getVoltageAngleLimitsIndex().containsKey(id)) {
+            throw new PowsyblException("The network " + network.getId()
                     + " already contains a voltage angle limit with the id '" + id + "'");
         }
         if (!Double.isNaN(lowLimit) && !Double.isNaN(highLimit) && lowLimit >= highLimit) {
             throw new IllegalStateException("Voltage angle low limit must be lower than the high limit.");
         }
 
-        VoltageAngleLimit voltageAngleLimit = new VoltageAngleLimitImpl(id, from, to, lowLimit, highLimit);
+        VoltageLevelExt voltageLevel1 = (VoltageLevelExt) from.getVoltageLevel();
+        VoltageLevelExt voltageLevel2 = (VoltageLevelExt) to.getVoltageLevel();
+        Ref<NetworkImpl> networkRef = AbstractIdentifiableAdder.computeNetworkRef(network, voltageLevel1, voltageLevel2);
```

Comment:
Strange to call this method from here, once again VoltageAngleLimit is very very close to an Identifiable...